File: JoyeriaBack/joyeria/src/main/java/com/sam/joyeria/anillo/AnilloRequest.java
package com.sam.joyeria.anillo;

import java.util.List;
import java.util.Map;

import com.sam.joyeria.anillo.model.Anillo;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
public class AnilloRequest {

	private int id;
	private int precio;
	private int idCategoria;
	private String descripcion;
	private String idAnillo;
	private String nombreAnillo;
	private int idCatalogo;
	private Map<String, String> catalogoImagenes;

	public AnilloRequest(Anillo anillo) {
		this.id = anillo.getId();
		this.precio = anillo.getPrecio();
		this.idCategoria = anillo.getIdCategoria();
		this.descripcion = anillo.getDescripcion();
		this.nombreAnillo = anillo.getNombreAnillo();
		this.idCatalogo = anillo.getIdCatalogo();
		
	}

}

File: Venta de Joyeria/app/src/app/models/anillo-request.ts
import { CategoriasAnillos } from "./categorias-anillos";

export class AnilloRequest {
    public  id:  number;
	public  precio: number;
	public  idCategoria: number;
	public  descripcion: string;
	public  idAnillo :string ;
	public  nombreAnillo : string ;
	public  idCatalogo : number ;
	public catalogoImagenes : Map<string, string>;
	public imagenMuestra: string;

}
